lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -215,6 +215,33 @@ def test_jumpto_id_invalid_location(self):
         response = self.client.get(jumpto_url)
         self.assertEqual(response.status_code, 404)
 
+    def test_jump_to_for_learner_with_staff_only_content(self):
```

Comment:
to add `ddt` to verify all the scenarios for the fix. `request` parameter can be mocked to set to a particular value for each case.

Reply:
have added `ddt` to test case. Could not test the masquerading scenario as we are getting `role` from course opaque key which are immutable, hence we cant set an attribute on it in test.